File: ainpc-core-plugin/src/main/java/ro/ainpc/ai/orchestration/AIOrchestrationRequest.java
package ro.ainpc.ai.orchestration;

import java.util.LinkedHashMap;
import java.util.Map;

public record AIOrchestrationRequest(
    AIUseCase useCase,
    String actorId,
    String playerName,
    String mechanicId,
    Map<String, String> context
) {
    public AIOrchestrationRequest {
        useCase = useCase != null ? useCase : AIUseCase.DIALOGUE_REPLY;
        actorId = valueOrEmpty(actorId);
        playerName = valueOrEmpty(playerName);
        mechanicId = valueOrEmpty(mechanicId);
        context = sanitize(context);
    }

    private static String valueOrEmpty(String value) {
        return value == null ? "" : value.trim();
    }

    private static Map<String, String> sanitize(Map<String, String> source) {
        if (source == null || source.isEmpty()) {
            return Map.of();
        }
        Map<String, String> sanitized = new LinkedHashMap<>();
        source.forEach((key, value) -> {
            String safeKey = valueOrEmpty(key);
            if (!safeKey.isBlank()) {
                sanitized.put(safeKey, valueOrEmpty(value));
            }
        });
        return Map.copyOf(sanitized);
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/ai/orchestration/AIOrchestrationRequest.kt
package ro.ainpc.ai.orchestration

import java.util.Collections
import java.util.LinkedHashMap

class AIOrchestrationRequest(
    useCase: AIUseCase?,
    actorId: String?,
    playerName: String?,
    mechanicId: String?,
    context: Map<String?, String?>?
) {
    private val useCaseValue = useCase ?: AIUseCase.DIALOGUE_REPLY
    private val actorIdValue = valueOrEmpty(actorId)
    private val playerNameValue = valueOrEmpty(playerName)
    private val mechanicIdValue = valueOrEmpty(mechanicId)
    private val contextValue = sanitize(context)

    fun useCase(): AIUseCase = useCaseValue

    fun actorId(): String = actorIdValue

    fun playerName(): String = playerNameValue

    fun mechanicId(): String = mechanicIdValue

    fun context(): Map<String, String> = contextValue

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is AIOrchestrationRequest) {
            return false
        }

        return useCaseValue == other.useCaseValue &&
            actorIdValue == other.actorIdValue &&
            playerNameValue == other.playerNameValue &&
            mechanicIdValue == other.mechanicIdValue &&
            contextValue == other.contextValue
    }

    override fun hashCode(): Int {
        var result = useCaseValue.hashCode()
        result = 31 * result + actorIdValue.hashCode()
        result = 31 * result + playerNameValue.hashCode()
        result = 31 * result + mechanicIdValue.hashCode()
        result = 31 * result + contextValue.hashCode()
        return result
    }

    override fun toString(): String =
        "AIOrchestrationRequest[useCase=$useCaseValue, actorId=$actorIdValue, " +
            "playerName=$playerNameValue, mechanicId=$mechanicIdValue, context=$contextValue]"

    companion object {
        private fun valueOrEmpty(value: String?): String = value?.trim().orEmpty()

        private fun sanitize(source: Map<String?, String?>?): Map<String, String> {
            if (source.isNullOrEmpty()) {
                return emptyMap()
            }

            val sanitized = LinkedHashMap<String, String>()
            for ((key, value) in source) {
                val safeKey = valueOrEmpty(key)
                if (safeKey.isNotBlank()) {
                    sanitized[safeKey] = valueOrEmpty(value)
                }
            }
            return Collections.unmodifiableMap(LinkedHashMap(sanitized))
        }
    }
}
